uniffi_meta/src/lib.rs

```diff
@@ -218,53 +223,7 @@ pub struct FnParamMetadata {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize)]
-pub enum Type {
-    U8,
-    U16,
-    U32,
-    U64,
-    I8,
-    I16,
-    I32,
-    I64,
-    F32,
-    F64,
-    Bool,
-    String,
-    Duration,
-    ForeignExecutor,
-    SystemTime,
-    Enum {
-        name: String,
-    },
-    Record {
-        name: String,
-    },
-    ArcObject {
-        object_name: String,
-        is_trait: bool,
-    },
-    CallbackInterface {
-        name: String,
-    },
-    Custom {
-        name: String,
-        builtin: Box<Type>,
-    },
-    Option {
-        inner_type: Box<Type>,
-    },
-    Vec {
-        inner_type: Box<Type>,
-    },
-    HashMap {
-        key_type: Box<Type>,
-        value_type: Box<Type>,
-    },
-}
-
-#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize)]
-pub enum Literal {
+pub enum LiteralMetadata {
```

Comment:
I think the literals are going to need to change some more, but this seems reasonable for now